samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -640,10 +640,25 @@ def _process_resolved_layers(
         "Map the resolved layers %s to configuration function %s.", resolved_layers, function_tf_resource.full_address
     )
     layers = []
+    does_refer_to_constant_values = False
+    does_refer_to_data_sources = False
     for resolved_layer in resolved_layers:
         # Skip ConstantValue layers reference, as it will be already handled by terraform plan command.
-        if isinstance(resolved_layer, ResolvedReference):
-            layers += _process_reference_layer_value(function_tf_resource, resolved_layer, tf_layers)
+        if isinstance(resolved_layer, ConstantValue):
```

Comment:
This is already done in another PR right? it could be removed.

Reply:
yes, It seems by mistake I merged this change https://github.com/aws/aws-sam-cli/pull/4387 to this one before I create the PR :(